### PR TITLE
oracle-instantclient, odpic: add aarch64

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

Tried to make n8n build again, failed, but this part works at least.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/oracle-instantclient/default.nix [hunk 1]
```diff
@@ -22,18 +22,32 @@ let
...
+    aarch64-linux = "19.10.0.0.0";
     x86_64-darwin = "19.3.0.0.0";
   }.${stdenv.hostPlatform.system} or throwSystem;
 
+  directory = {
+    x86_64-linux = "1916000";
+    aarch64-linux = "191000";
+    x86_64-darwin = "193000";
+  }.${stdenv.hostPlatform.system} or throwSystem;
+
```
Comment (flokli) [inline pkgs/development/libraries/oracle-instantclient/default.nix:35]:
```suggestion
```

File: pkgs/development/libraries/oracle-instantclient/default.nix [hunk 2]
```diff
@@ -50,24 +64,25 @@ let
...
     "instantclient-${component}-${arch}-${version}" +
     (optionalString (rel != "") "-${rel}") +
-    (optionalString (arch == "linux.x64" || arch == "macos.x64") "dbru") + # ¯\_(ツ)_/¯
-    ".zip";
+    "dbru.zip"; # ¯\_(ツ)_/¯
 
   # fetcher for the non clickthrough artifacts
   fetcher = srcFilename: hash: fetchurl {
-    url = "https://download.oracle.com/otn_software/${shortArch}/instantclient/193000/${srcFilename}";
+    url = "https://download.oracle.com/otn_software/${shortArch}/instantclient/${directory}/${srcFilename}";
```
Comment (flokli) [inline pkgs/development/libraries/oracle-instantclient/default.nix:85]:
```suggestion
    url = "https://download.oracle.com/otn_software/${shortArch}/instantclient/${builtins.replaceStrings ["."] [""] version}/${srcFilename}";
```

File: pkgs/development/libraries/oracle-instantclient/default.nix (hunk 1 above)
Comment (K900) [reply pkgs/development/libraries/oracle-instantclient/default.nix:35]: That would be too easy. There's a 0 missing in 191000.

Comment (flokli) [reply pkgs/development/libraries/oracle-instantclient/default.nix:35]: ugh.

**Pull request merged**